Update CSP rules to allow newer DAP (GA 4)

zachmargolis opened the pull request:
See [Slack thread](https://gsa-tts.slack.com/archives/C5E7EJWF7/p1709755279457439) for context

aduth commented:
I think there's a few more directives that need to be added for `img-src` and `connect-src` as well:

https://developers.google.com/tag-platform/security/guides/csp#google_analytics_4_google_analytics

zachmargolis commented:
> I think there's a few more directives that need to be added for `img-src` and `connect-src` as well:
> 
> https://developers.google.com/tag-platform/security/guides/csp#google_analytics_4_google_analytics

Thanks! Added in dbf972905ced7fc78e4c418da6f9b87e9830f5a1, and added specs with a link to those docs too

aduth reviewed (approved):
LGTM! 👍
